File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> hashy = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int comp = target - nums[i];
            if (hashy.containsKey(nums[i])) {
                return new int[] {hashy.get(nums[i]), i};
            } 
            hashy.put(comp, i);
        }   
        return null;
    }
}

File: 1-two-sum/two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        numDict = {}
        n = len(nums)

        for i in range(n):
            comp = target - nums[i]
            if comp in numDict:
                return [numDict[comp], i]
            numDict[nums[i]] = i

        return []

        